DOCUMENTATION.md

```diff
@@ -0,0 +1,127 @@
+```python
+client.start()
+```
+
+### stop
+Stops the MQTT client and also disconnect it.
+```python
+client.stop()
+```
+
+### Using Callbacks
+
+The callback functions are functions which are executed when a trigger event happens. They should be set right after the MQTT client creation so that they don't miss any event.
+
+#### setUplinkCallback
+Set the callback function, to be called when an uplink message is received.
+```python
+client.setUplinkCallback(uplinkCallback)
+```
+
+##### uplinkCallback
+The callback function must be declared in your script following this structure:
+* `uplinkCallback(msg, client)`
+  * `msg`: **JSON object**  the message received by the client
+  * `client`: **object**  the client from which the callback is executed are calling
+
+On each message reception, you should see **receving message from** in the console, and the callback will be executed.
+
+#### setConnectCallback
+Set the connection callback function to be executed when the client connect to the broker.
+```python
+client.setConnectCallback(connectCallback)
+```
+##### connectCallback
+- `connectCallback(res, client)`: the function which will be executed on connection to the broker.
+  - `res`: **int**  the result of the connection. If it's 0, it went well. If not, it means the connection failed.
+  - `client`: **object**  the TTN client from which we call the callback.
+
+#### setDownlinkCallback
+Set the downlink callback function, with actions to execute when a downlink message is sent.
+```python
+client.setDownlinkCallback(downlinkCallback)
+```
+##### downlinkCallback
+- `downlinkCallback(mid, client)`: the function which will be the new publish behavior for our MQTT client.
+  - `mid`: **int**  it matches the mid variable returned from the publish call to allow sent messages to be tracked.
+  - `client`: **object**  the TTN client from which we call the callback.
+
+#### setCloseCallback
+Set the callback to be executed when the connection to the TTN broker is closed.
+```python
+client.setCloseCallback(closeCallback)
+```
+##### closeCallback
+- `closeCallback(res, client)`: the function which will be executed when the connection is closed.
+  - `res`: **int**  the result of the disconnection. If it's 0, it went well. If not, it means the disconnection was unexpected.
+  - `client`: **object**  the TTN client from which we call the callback.
+
+### send
+Publishes a message to the MQTT broker.
+```python
+client.send(deviceID, payload, [port], [confirmation], [schedule])
+```
+- `deviceID`: **string**  the ID of the device you wish to send the message to.
+- `payload`: the payload of the message to be published to the broker. It can be an hexadecimal **string** like `AQ==` (this will send the raw payload `00` to your device) or an object with several fields following the **JSON** standard.
```

Comment:
+ This is not hexadecimal, this is base64
+ Can you add more details to the object option, and how they relate to payload functions?